server.go

```diff
@@ -311,3 +314,161 @@ func (s *server) ListLimits(ctx context.Context,
 		Limits:       rpcLimits,
 	}, nil
 }
+
+func (s *server) ListForwardingHistory(ctx context.Context,
+	req *circuitbreakerrpc.ListForwardingHistoryRequest) (
+	*circuitbreakerrpc.ListForwardingHistoryResponse, error) {
+
+	var (
+		// By default query from the epoch until now.
+		startTime = time.Time{}
+		endTime   = time.Now()
+	)
+
+	if req.StartTimeNs != 0 {
+		startTime = time.Unix(0, req.StartTimeNs)
+	}
+
+	if req.EndTimeNs != 0 {
+		endTime = time.Unix(0, req.EndTimeNs)
+	}
+
+	if startTime.After(endTime) {
+		return nil, fmt.Errorf("start time: %v after end time: %v", startTime,
+			endTime)
+	}
+
+	// To report report on channel utilization, we need to get all the HTLCs that were in
+	// flight at the beginning of our window. If our start time is zero, we won't have anything
+	// in flight.
+	inFlight, err := s.db.ListInFlightAt(ctx, startTime)
+	if err != nil {
+		return nil, err
+	}
+
+	htlcs, err := s.db.ListForwardingHistory(ctx, startTime, endTime)
```

Comment:
Hmm yes, it seems better to let the rpc be more low level with just the ability to query forwarding data based on add and resolve time, and then do what's in this commit externally. In a jamming-specific cli tool that makes the rpc call, or some other post-processing script outside of circuitbreaker?

Reply:
Yeah, it felt pretty heavy even when I was writing it. Going to cut it out completely and just do forwarding data - I can re-use the code in a post-processing tool.

Reply:
Yes, happier now!